src/libraries/System.Text.Json/tests/Serialization/NumberHandlingTests.cs

```diff
@@ -513,6 +514,33 @@ public MyClassWithNumbers(ulong? @ulong, List<float> listOfFloats)
             }
         }
 
+
+        [Fact]
+        public static void Number_AsObjectWithParameterizedCtor_PropHasAttribute()
+        {
+            string json = @"{""ListOfFloats"":[""1""]}";
+            // Strict handling on property overrides loose global policy.
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<MyClassWithNumbers_PropsHasAttribute>(json, s_optionReadFromStr));
+
+            // Serialize
+            json = @"{""ListOfFloats"":[1]}";
+            MyClassWithNumbers_PropsHasAttribute obj = JsonSerializer.Deserialize<MyClassWithNumbers_PropsHasAttribute>(json);
+
+            // Number serialized as JSON number due to strict handling on property which overrides loose global policy.
+            Assert.Equal(json, JsonSerializer.Serialize(obj, s_optionReadAndWriteFromStr));
```

Comment:
> which overrides loose global policy.

It looks like s_optionsReadAndWriteFromStr just adds a couple custom converters and does not change the global number format